e2e-tests/tests/test_jolteon_consensus_rpc.py

```diff
@@ -0,0 +1,424 @@
+                assert False, "Failed to retrieve replica state"
+            
+            replica_state = result['result']
+            logger.info(f"Replica state: {json.dumps(replica_state, indent=2)}")
+            
+            # Verify required fields exist
+            required_fields = ['r_curr', 'r_vote', 'r_lock', 'qc_high', 'tc_last', 'storage_block_count']
+            for field in required_fields:
+                assert field in replica_state, f"Missing required field: {field}"
+            
+            # Verify data types and basic sanity checks
+            assert isinstance(replica_state['r_curr'], int), "r_curr should be an integer"
+            assert isinstance(replica_state['r_vote'], int), "r_vote should be an integer"
+            assert isinstance(replica_state['r_lock'], int), "r_lock should be an integer"
+            assert isinstance(replica_state['storage_block_count'], int), "storage_block_count should be an integer"
+            
+            # Verify round numbers are reasonable
+            assert replica_state['r_curr'] >= 0, "r_curr should be non-negative"
+            assert replica_state['r_vote'] >= 0, "r_vote should be non-negative"
+            assert replica_state['r_lock'] >= 0, "r_lock should be non-negative"
+            
+            # Verify QC structure
+            qc_high = replica_state['qc_high']
+            assert 'block_hash' in qc_high, "QC should have block_hash"
+            assert 'round' in qc_high, "QC should have round"
+            assert 'vote_count' in qc_high, "QC should have vote_count"
+            
+            logger.info(f"✅ Replica state test passed:")
+            logger.info(f"  Current round: {replica_state['r_curr']}")
+            logger.info(f"  Last voted round: {replica_state['r_vote']}")
+            logger.info(f"  Locked round: {replica_state['r_lock']}")
+            logger.info(f"  Storage blocks: {replica_state['storage_block_count']}")
+            
+        except Exception as e:
+            logger.error(f"Error testing replica state: {e}")
+            raise
+
+    @mark.test_key('JOLTEON-RPC-002')
+    def test_round_progression(self, api: BlockchainApi, config: ApiConfig):
+        """Test that rounds are progressing correctly
+        
+        This test monitors round progression over time to verify
+        that the Jolteon consensus is making progress.
+        """
+        logger.info("Starting Jolteon round progression test")
+        
+        # Get initial round state
+        initial_result = api.substrate.rpc_request("jolteon_getRoundInfo", [])
+        if not initial_result or 'result' not in initial_result:
+            assert False, "Failed to get initial round info"
+        
+        initial_round = initial_result['result']
+        logger.info(f"Initial round state: {initial_round}")
+        
+        # Wait for round progression
+        wait_time = 30  # Wait 30 seconds for round advancement
```

Comment:
Instead of 30 seconds can we find the protocol parameter and set it here? It should be called something like max round time, or round timeout.